source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -296,64 +296,26 @@ ShaderPtr GlslShaderGenerator::generate(const string& name, ElementPtr element,
 
 void GlslShaderGenerator::emitVertexStage(const ShaderGraph& graph, GenContext& context, ShaderStage& stage) const
 {
-    HwResourceBindingContextPtr resourceBindingCtx = context.getUserData<HwResourceBindingContext>(HW::USER_DATA_BINDING_CONTEXT);
+    HwResourceBindingContextPtr resourceBindingCtx = getResourceBindingContext(context);
 
-    // Add directives
-    emitLine("#version " + getVersion(), stage, false);
+    emitDirectives(context, stage);
```

Comment:
Most of the merge logic differences are around the pulling out of methods which confliced with what was already there. Please make sure I didn't mess up any logic as I had to hand-edit this a bunch. Thanks.